pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -831,6 +831,16 @@ public interface ConsumerBuilder<T> extends Cloneable {
      */
     ConsumerBuilder<T> autoScaledReceiverQueueSizeEnabled(boolean enabled);
 
+    /**
+     * Sets the {@link SubscriptionIsolationLevel} for the consumer.
+     *
+     * @param subscriptionIsolationLevel If READ_COMMITTED is selected, the Consumer can only consume all transactional messages which have been committed,
+     *                                   else if READ_UNCOMMITTED is selected, the Consumer can consume all messages, even transactional messages which have been aborted.
+     *                                   Note that this is a subscription dimension configuration, and all consumers under the same subscription need to be configured with the same IsolationLevel.
```

Comment:
These lines are too long, so the check of `check style` could not pass. Could you run `mvn clean install -Dskiptests` first to ensure the check can passed?

Reply:
> @codelipenghui
> Hi, could you please help with the review as soon as possible? We are currently very urgent.
> Thank you for your time

Since you are urgent, you should also focus on the failed CIs.

<img width="893" alt="Screenshot 2023-10-26 at 22 30 49" src="https://github.com/apache/pulsar/assets/25195800/b4c828ff-f9e7-4341-8cba-16466949b38e">

Reply:
You can set up your IDE by following this doc: https://pulsar.apache.org/contribute/setup-ide/

Reply:
The [doc](https://kafka.apache.org/0110/javadoc/index.html?org/apache/kafka/clients/consumer/KafkaConsumer.html) of Kafka also can be referred to.

<img width="1506" alt="image" src="https://github.com/apache/pulsar/assets/12592133/23728537-db01-44a9-828e-a5f9adc6cbe5">

Reply:
@codelipenghui @hzh0425 

In the current PR, the implementation is different with Kafka:
-  Kafka: the consumer will receive all messages, even if the messages have been aborted.
- Current PR:
  - the consumer will receive committed messages;
  - the consumer will receive the messages that state `transaction on-gonging`, once the transaction was aborted, these messages could not be received.

Reply:
We have discussed this question in the meeting. At that time, the consensus was that not receiving aborted messages was a better solution.

Reply:
The original intention of using read uncommitted is to reduce read and write latency, not to read error messages

Reply:
There are differences between Kafka and Pulsar in filtering the aborted message. 

Kafka filters the aborted message on the client side, so if the client uses `READ_UNCOMMITED`, the client can't feel the ongoing transaction, the client is only able to filter the markers. If we need to filter the abort message on the kafka client, it requires to response of the `abortedTransactions` data each fetch, which will reduce the consumption performance.

Pulsar has different implements for filter parts, the filter logic is on the Broker side. If it is to reduce read and write latency, it is okay.

But I think reading all messages just like Kafka is beneficial in the following cases:
In the event of issues or during development and testing phases, having access to all messages, including uncommitted ones, can be valuable for debugging and troubleshooting.